src/coreclr/jit/llvm.cpp

```diff
@@ -1127,6 +1259,12 @@ Value* Llvm::localVar(GenTreeLclVar* lclVar)
         llvmRef = _localsMap->at({lclNum, ssaNum});
     }
 
+    // implicit truncating from long to int
+    if (llvmRef->getType() == Type::getInt64Ty(_llvmContext) && getLlvmTypeForVarType(lclVar->TypeGet()) == Type::getInt32Ty(_llvmContext))
```

Comment:
```suggestion
    if (llvmRef->getType() == Type::getInt64Ty(_llvmContext) && lclVar->TypeIs(TYP_INT))
```
Seems simpler to use the tree type directly.

I think this should also be handled in lowering, though it does not matter much I guess.

Reply:
thanks,